File: src/LogParse.java
import java.util.*;
import java.io.*;

public class LogParse {
    public static void main(String[] args) {
        try {
            System.out.println(Arrays.toString(args));
            Scanner s;
            if(args.length == 0) {
                s = new Scanner(new File("Player.log"));
            } else {
                s = new Scanner(new File(args[0]));
            }

            parse(s);
        } catch(Exception e) {
            System.out.println("failed to get file (probably)");
            System.out.println(e);
            StringWriter sw = new StringWriter();
            PrintWriter pw = new PrintWriter(sw);
            e.printStackTrace(pw);
            String sStackTrace = sw.toString(); // stack trace as a string
            System.out.println(sStackTrace);
        }
    }

    public static void parse(Scanner s) {
        String currentTileset = "";
        String currentMap = "";
        while(s.hasNextLine()) {
            String line = s.nextLine();
            String line_substr = "";
//            try {
                if(line.substring(6, 18).equals("Done Loading")) {
                    String newMap = line.substring(line.indexOf(']') + 2, line.lastIndexOf('[') - 1);
                    if(!newMap.equals(currentMap)) {
                        System.out.println("round change!");
                        currentMap = newMap;
                        currentTileset = line.substring(line.indexOf('[') + 1, line.indexOf(']'));
                        System.out.println("new tileset: " + currentTileset + "\nnew map: " + currentMap);
                    }
                }

                if(line.substring(0, 6).equals("Killer")) {
                    String killerSide = line.substring(line.indexOf(':') + 2, line.indexOf(','));
                    String victimSide = line.substring(line.lastIndexOf(':') + 2);
                    String killInfoLine = s.nextLine();
                    System.out.println(killerSide + " killed " + victimSide + ": \n" + killInfoLine);
                }
//            } catch (Exception e) {
//
//            }
        }
    }

    public static Team getTeam(String raw) {
        return switch(raw) {
            case "ATTACKER" -> Team.Attacker;
            case "DEFENDER" -> Team.Defender;
            case "NONE" -> Team.None;
            default -> Team.Unknown;
        };
    }

    public static enum Team {
        Attacker,
        Defender,
        None,
        Unknown,
    }
}

File: src/LogParse.kt
import java.io.File
import java.io.PrintWriter
import java.io.StringWriter
import java.util.*

object LogParse {
    @JvmStatic
    fun main(args: Array<String>) {
        try {
//            println(Arrays.toString(args)
            val s: Scanner
            s = if (args.size == 0) {
                Scanner(File("Player.log"))
            } else {
                Scanner(File(args[0]))
            }
            parse(s)
        } catch (e: Exception) {
            println("failed to get file (probably)")
            println(e)
            val sw = StringWriter()
            val pw = PrintWriter(sw)
            e.printStackTrace(pw)
            val sStackTrace = sw.toString() // stack trace as a string
            println(sStackTrace)
        }
    }

    fun parse(s: Scanner) {
//        var currentTileset = ""
//        var currentMap = ""
        var match = MatchLog()
        var currentRound = RoundLog()
        match.addRound(currentRound)
        while (s.hasNextLine()) {
            val line = s.nextLine()
            try {
                if (line.contains("Done Loading Game Level")) {
                    val newMap = line.substring(line.indexOf(']') + 2, line.lastIndexOf('[') - 1)
                    if (newMap != currentRound.map) {
//                        println("round change!")
                        val tileset = line.substring(line.indexOf('[') + 1, line.indexOf(']'))
                        currentRound = RoundLog(tileset, newMap)
                        match.addRound(currentRound)
//                        println("new tileset: $tileset\nnew map: $newMap")
                    }
                }
                if (line.substring(0, 6) == "Killer") {
                    val killerSide = line.substring(line.indexOf(':') + 2, line.indexOf(','))
                    val victimSide = line.substring(line.lastIndexOf(':') + 2)
                    var killInfoLine = s.nextLine()
//                    println("$killerSide killed $victimSide")

                    killInfoLine = killInfoLine.replace(Regex("<[^<||^>]*>"), "<>");
//                    var killInfoList = ("<$killInfoLine").split("((?<=<)|(?=>))".toRegex())
                    var killInfoList = ("<$killInfoLine").split("<>".toRegex())
                    killInfoList = killInfoList.filterIndexed { ix, element ->
                        ix != 0 && element.isNotBlank()
                    }

                    var killerName: String? = killInfoList[0].trim();
                    var victimName: String? = killInfoList[2].trim();

                    val killLog = KillLog(if(killerName!=null) killerName else "unknown", killerSide,
                            if(victimName!=null) victimName else "unknown", victimSide)
                    currentRound.addKill(killLog)
//                    println(killLog)
//                    TODO()
//                    currentRound.addKill(KillLog())
                }
            } catch (e: Exception) {
            }
        }
        println(match)
    }
}